grapher/stackedCharts/StackedDiscreteBarChart.tsx

```diff
@@ -0,0 +1,424 @@
+            .padTop(this.legend.height)
+    }
+
+    @computed private get selectionArray() {
+        return makeSelectionArray(this.manager)
+    }
+
+    @computed private get items(): Item[] {
+        const entityNames = this.selectionArray.selectedEntityNames
+        const items = entityNames.map((entityName) => ({
+            label: entityName,
+            bars: excludeUndefined(
+                this.series.map((series) => {
+                    const point = series.points.find(
+                        (point) => point.position === entityName
+                    )
+                    if (!point) return undefined
+                    return {
+                        point,
+                        color: series.color,
+                        seriesName: series.seriesName,
+                    }
+                })
+            ),
+        }))
+        return sortBy(items, (item) => {
+            const lastPoint = last(item.bars)?.point
+            if (!lastPoint) return 0
+            return lastPoint.valueOffset + lastPoint.value
+        }).reverse()
+    }
+
+    @computed private get barHeight() {
+        return (0.8 * this.innerBounds.height) / this.items.length
+    }
+
+    @computed private get barSpacing() {
+        return this.innerBounds.height / this.items.length - this.barHeight
+    }
+
+    // legend props
+
+    @computed get legendPaddingTop(): number {
+        return this.baseFontSize
+    }
+
+    @computed get legendX(): number {
+        return this.bounds.x
+    }
+
+    @computed get categoryLegendY(): number {
+        return this.bounds.top
+    }
+
+    @computed get legendWidth(): number {
+        return this.bounds.width
+    }
+
+    @computed get legendAlign(): LegendAlign {
+        return LegendAlign.left
+    }
+
+    @computed get fontSize(): number {
+        return this.baseFontSize
+    }
+
+    @computed get categoricalLegendData(): CategoricalBin[] {
+        return this.series.map((series, index) => {
+            return new CategoricalBin({
+                index,
+                value: series.seriesName,
+                label: series.seriesName,
+                color: series.color,
+            })
+        })
+    }
+
+    @action.bound onLegendMouseOver(bin: CategoricalBin) {
+        this.focusSeriesName = bin.value
+    }
+
+    @action.bound onLegendMouseLeave() {
+        this.focusSeriesName = undefined
+    }
+
+    @computed private get legend(): HorizontalCategoricalColorLegend {
+        return new HorizontalCategoricalColorLegend({ manager: this })
+    }
+
+    componentDidMount() {
+        exposeInstanceOnWindow(this)
+    }
+
+    componentDidUpdate() {
+        // Animating the bar width after a render ensures there's no race condition, where the
+        // initial animation (in componentDidMount) did override the now-changed bar width in
+        // some cases. Updating the animation with the updated bar widths fixes that.
+    }
```

Comment:
I can see how you sneakily copied that over from `DiscreteBarChart` ;)

Reply:
😶